JSDemos/Demos/List/ListSelection/React/App.js

```diff
@@ -14,82 +14,66 @@ const dataSource = new ArrayStore({
+        setSelectedItemKeys(args.value);
+      }
     }
-  }
+  }, [selectionMode, selectedItemKeys, setSelectedItemKeys]);
+
+  const onSelectionModeChange = React.useCallback((args) => {
+    setSelectionMode(args.value);
+  }, [setSelectionMode]);
 
-  render() {
-    return (
-      <React.Fragment>
-        <div className="widget-container">
-          <List
-            dataSource={dataSource}
-            height={400}
-            showSelectionControls={true}
-            selectionMode={this.state.selectionMode}
-            selectAllMode={this.state.selectAllMode}
-            selectedItemKeys={this.state.selectedItemKeys}
-            onOptionChanged={this.onSelectedItemKeysChange}>
-          </List>
-          <div className="selected-data">
-            <span className="caption">Selected IDs:</span>
-            <span>{this.state.selectedItemKeys.join(', ')}</span>
-          </div>
+  const onSelectAllModeChange = React.useCallback((args) => {
+    setSelectionMode(args.value);
+  }, [setSelectAllMode]);
+
+  return (
+    <React.Fragment>
+      <div className="widget-container">
+        <List
+          dataSource={dataSource}
+          height={400}
+          showSelectionControls={true}
+          selectionMode={selectionMode}
+          selectAllMode={selectAllMode}
+          selectedItemKeys={selectedItemKeys}
+          onOptionChanged={onSelectedItemKeysChange}>
+        </List>
+        <div className="selected-data">
+          <span className="caption">Selected IDs: </span>
+          <span>{ selectedItemKeys.join(', ') }</span>
         </div>
-        <div className="options">
-          <div className="caption">Options</div>
-          <div className="option">
-            <span>Selection Mode</span>
+      </div>
+      <div className="options">
+        <div className="caption">Options</div>
+        <div className="option">
+          <span>Selection Mode</span>
             &nbsp;
-            <SelectBox
-              items={selectionModes}
-              value={this.state.selectionMode}
-              onValueChanged={this.onSelectionModeChange}>
-            </SelectBox>
-          </div>
-          <div className="option">
-            <span>Select All Mode</span>
+          <SelectBox
+            items={selectionModes}
+            value={selectionMode}
+            onValueChanged={onSelectionModeChange}>
+          </SelectBox>
+        </div>
+        <div className="option">
+          <span>Select All Mode</span>
             &nbsp;
-            <SelectBox
-              disabled={this.state.selectionMode !== 'all'}
-              items={selectAllModes}
-              value={this.state.selectAllMode}
-              onValueChanged={this.onSelectAllModeChange}>
-            </SelectBox>
-          </div>
+          <SelectBox
+            disabled={selectionMode !== 'all'}
+            items={selectAllModes}
+            value={selectAllMode}
+            onValueChanged={onSelectAllModeChange}>
```

Comment:
In react should be onValueChange https://js.devexpress.com/Documentation/Guide/React_Components/State_Management/?search=onValueChange

Reply:
It is strange that is no public option onValueChange. I passed this to our techwriters and they have a card already to fix it misunderstanding. 

Fixed in https://github.com/DevExpress/devextreme-demos/pull/2519/commits/9c54b26550612601382a816f984eed2a50931994